### ISSUE TITLE
Jitting a dropout module yields a constant value

### ISSUE THREAD

**Issue by bitsandscraps** [google/flax#4539]
### System information
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 22.04.4
- Flax, jax, jaxlib versions (obtain with `pip show flax jax jaxlib`: flax=0.10.2, jax=0.4.33, jaxlib=0.4.33
- Python version: 3.11.11
- GPU/TPU model and memory: NVIDIA Titan Xp
- CUDA version (if applicable):

### Problem you have encountered:

Jitting a dropout module yields a constant value. Is this intended?

### What you expected to happen:

I think it should output a different value each time you call it.

### Logs, error messages, etc:

Refer to the colab file.

### Steps to reproduce:

https://colab.research.google.com/drive/1KuUdrN6F7grguXhxByA-BRRpZTBHYPfs?usp=sharing

**Comment by bitsandscraps** [comment]
If you use
```
from flax import nnx
from jax import numpy as jnp

@nnx.jit
def forward(model, x):
    return model(x)

module = nnx.Dropout(0.5, rngs=nnx.Rngs(0))
forward(module, jnp.ones((3, 5)))
```
then everything works fine.
